### PR TITLE
Asset check selection

### PR DESCRIPTION
**Pull request by johannkm**
Enable executing specific checks by passing an `asset_check_selection` through an incredible number of layers.

Wrapping asset_selection and asset_check_selection would make type signatures a bit more concise, but I think there'd be a similar amount of distributed logic. This certainly doesn't help the situation, but I don't think it hurts it tooo much...

The initial intention is to support two cases from the front end:
- continue supporting the behavior that clicking materialize on an asset both materializes and runs checks. This looks like:
  - asset_selection is None, and asset_check_selection is None: execute all assets and checks in a job
  - asset_selection is [some asset], and asset_check_selection is None: execute some_asset and all checks on it
- support running an individual check
  - asset_selection is [], and asset_check_selection is [my_check]: execute only my_check

In the future we'll support
- materialize an asset without running checks
  - asset_selection is [my_asset], and asset_check_selection is []
- arbitrary combinations of checks and assets

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16356** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16356" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16356?utm_source=stack-comment).

File: js_modules/dagster-ui/packages/ui-core/src/graphql/schema.graphql [hunk 1]
```diff
@@ -2100,6 +2100,12 @@ input JobOrPipelineSelector {
   repositoryLocationName: String!
   solidSelection: [String!]
   assetSelection: [AssetKeyInput!]
+  assetCheckSelection: [AssetCheckHandleInput!]
```
Comment (johannkm) [inline js_modules/dagster-ui/packages/ui-core/src/graphql/schema.graphql:2103]: @bengotow this is the schema for now

**Review by schrockn — DISMISSED** [review]
Oh boy.

So I took a run and consolidating `asset_selection` and `op_selection` into a single object to try to contain this problem because I threading those things in parallel through the code base doesn't really make any sense. This makes that worse.

While I do not think we should add a user-facing abstraction this diff, I would much rather see a new object that contains an asset selection and asset check selection to avoid yet another set of parallel arguments. We can then take it as an action item to merge op selection in _that_ object. How difficult do you think it would be to do that in this PR?

**Comment by schrockn** [comment]
One request:

Please exhaustively list the _public_ APIs we are adding this to

**Comment by johannkm** [comment]
gql:
JobOrPipelineSelector
PipelineSelector
(Seems like we shouldn’t need both, posted in help-eng)

New field that’s accessible on DagsterRun - this isn't marked public but I can imagine someone using for e.g. an alerting sensor

Not in this diff, but will need it on:
RunRequest
define_asset_job

**Review by schrockn — APPROVED** [review]
Per offline discussion, going to move forward with this.

**Pull request merged**